tests/linen/linen_linear_test.py

```diff
@@ -144,7 +144,7 @@ def _counter_init(rng, shape, dtype, state):
     np.testing.assert_allclose(y, target)
 
   @parameterized.parameters([((-2, 3), (), 'bijk,jklm->bilm'),
-                             ((3, -2), (), 'bijk,kjlm->bilm'),
+                             ((3, -2), (), 'bijk,jklm->bilm'),
```

Comment:
So our testcase was actually testing wrong behavior? That's odd 😄

Reply:
No this was correct. We are free to pick the order of the kernel dimensions. But I made it invariant wrt to the order in which you pass the axes. I think this is more consistent and it makes the implementation easier.